test/collect_coverage_api_test.dart

```diff
@@ -98,12 +98,40 @@ void main() {
       expect(sampleCoverageData['funcHits'], isNotEmpty);
     }
   });
+
+  test('collect_coverage_api with branch coverage', () async {
+    final json = await _collectCoverage(branchCoverage: true);
+    expect(json.keys, unorderedEquals(<String>['type', 'coverage']));
+    expect(json, containsPair('type', 'CodeCoverage'));
+
+    final coverage = json['coverage'] as List;
+    expect(coverage, isNotEmpty);
+
+    final sources = coverage.cast<Map>().fold(<String, List<Map>>{},
+        (Map<String, List<Map>> map, value) {
+      final sourceUri = value['source'] as String;
+      map.putIfAbsent(sourceUri, () => <Map>[]).add(value);
+      return map;
+    });
+
+    if (platformVersionCheck(2, 17)) {
+      // Dart VM versions before 2.17 don't support branch coverage.
+      for (var sampleCoverageData in sources[_sampleAppFileUri]!) {
```

Comment:
it would be better to avoid for loops in the tests. If this fails the message wont give much information.

We could do something like

expect(sources[_sampleAppFileUri], everyElement(containsPair('branchHits', isNotEmpty)));

Reply:
Done